Sources/Commands/Workspace.swift

```diff
@@ -345,38 +412,10 @@ public class Workspace {
         // delegate of what is happening.
         delegate.fetchingMissingRepositories(missingURLs)
 
-        // Perform dependency resolution using the constraint set induced by the active checkouts.
-        var constraints = [RepositoryPackageConstraint]()
+        let constraints = computePackageConstraints(rootManifest: rootManifest, dependencies: currentExternalManifests)
 
-        // First, add the root package constraints.
-        constraints += rootManifest.package.dependencies.map{
-            RepositoryPackageConstraint(container: RepositorySpecifier(url: $0.url), versionRequirement: .range($0.versionRange))
-        }
-
-        // Add constraints to pin to *exactly* all the checkouts we have.
-        //
-        // FIXME: We may need a better way to tell the resolution algorithm that
-        // certain repositories are pinned to the current checkout. We might be
-        // able to do that simply by overriding the view presented by the
-        // repository container provider.
-        for externalManifest in currentExternalManifests {
-            let specifier = RepositorySpecifier(url: externalManifest.url)
-            let managedDependency = dependencyMap[specifier]!
-
-            // If we know the manifest is at a particular version, use that.
-            if let version = managedDependency.currentVersion {
-                // FIXME: This is broken, successor isn't correct and should be eliminated.
-                constraints.append(RepositoryPackageConstraint(container: specifier, versionRequirement: .range(version..<version.successor())))
-            } else {
-                // FIXME: Otherwise, we need to be able to constraint precisely to the revision we have.
-                fatalError("FIXME: Unimplemented.")
-            }
-        }
-
-        // Solve for additional checkouts we require.
-        let resolverDelegate = WorkspaceResolverDelegate()
-        let resolver = DependencyResolver(containerProvider, resolverDelegate)
-        let result = try resolver.resolve(constraints: constraints)
+        // Perform dependency resolution using the constraint set induced by the active checkouts.
+        let result = try resolveDependencies(constraints: constraints)
```

Comment:
I'd like it if we could merge some of the code below with the same kind of code in `updateDependencies`. If we have the base method return the list of all the modifications made, then we can use that to do both the fatalError check (that we didn't change something unexpectedly) and to know which new manifests we need to grab.